Required minimum typer version

> E               AssertionError: List types with complex sub-types are not currently supported


upgrading fixed it:

```
pip install typer --upgrade
Requirement already satisfied: typer in /home/moose/.pyenv/versions/3.8.3/lib/python3.8/site-packages (0.12.3)
Collecting typer
  Downloading typer-0.15.1-py3-none-any.whl.metadata (15 kB)
Requirement already satisfied: click>=8.0.0 in /home/moose/.pyenv/versions/3.8.3/lib/python3.8/site-packages (from typer) (8.1.7)
Requirement already satisfied: typing-extensions>=3.7.4.3 in /home/moose/.pyenv/versions/3.8.3/lib/python3.8/site-packages (from typer) (4.12.2)
Requirement already satisfied: shellingham>=1.3.0 in /home/moose/.pyenv/versions/3.8.3/lib/python3.8/site-packages (from typer) (1.5.4)
Requirement already satisfied: rich>=10.11.0 in /home/moose/.pyenv/versions/3.8.3/lib/python3.8/site-packages (from typer) (13.9.4)
Requirement already satisfied: markdown-it-py>=2.2.0 in /home/moose/.pyenv/versions/3.8.3/lib/python3.8/site-packages (from rich>=10.11.0->typer) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /home/moose/.pyenv/versions/3.8.3/lib/python3.8/site-packages (from rich>=10.11.0->typer) (2.18.0)
Requirement already satisfied: mdurl~=0.1 in /home/moose/.pyenv/versions/3.8.3/lib/python3.8/site-packages (from markdown-it-py>=2.2.0->rich>=10.11.0->typer) (0.1.2)
Downloading typer-0.15.1-py3-none-any.whl (44 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.9/44.9 kB 810.8 kB/s eta 0:00:00
Installing collected packages: typer
  Attempting uninstall: typer
    Found existing installation: typer 0.12.3
    Uninstalling typer-0.12.3:
      Successfully uninstalled typer-0.12.3
Successfully installed typer-0.15.1
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,7 +32,7 @@ classifiers = [
 
 dependencies = [
     "pypdf>=3.8.2",
-    "typer>=0.9.0",
+    "typer>=0.12.4",
     "pillow",
     "pydantic",
     "rich",

